pkgs/unified_analytics/lib/src/analytics.dart

```diff
@@ -187,6 +187,10 @@ abstract class Analytics {
         enableAsserts: true,
       );
 
+  /// The shared identifier for Flutter and Dart related tooling using
+  /// package:unified_analytics.
+  String get clientId;
```

Comment:
is this addition going to break existing clients? if so, we better bump the major version.

Reply:
Ah good call, seems like the analysis server is implementing our NoOp instance so I will bump the version to 4.0.0

https://github.com/dart-lang/sdk/blob/main/pkg/analysis_server/test/src/analytics/analytics_manager_test.dart#L559-L564